server/plugin.ts

```diff
@@ -97,6 +105,10 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
         plugins: [opensearchSecurityConfigurationPlugin, opensearchSecurityPlugin],
       }
     );
+    if (dataSourceEnabled) {
```

Comment:
can you add a brief comment here to help understand what does this block do?

Reply:
Similar to https://github.com/opensearch-project/security-dashboards-plugin/pull/1818/files#diff-006ee2c1d9e1d5af281d223734f25cceadca71a31ce838157231f2dbbdf63addR105, in order to use the client provided by data source plugin, we need to add the config so that the client knows how to talk to plugin backend by registering the schema. The registerCustomApiSchema essentially does the same for the data source client as in line 105 above